.github/workflows/template-ledger-epoch-snapshots.yml

```
@@ -178,71 +100,43 @@ jobs:
-
-          echo "Auto-resolved RUN_UNTIL_TARGET_EPOCH=$RUN_UNTIL_TARGET_EPOCH"
-
-          # Make sure to export RUN_UNTIL_TARGET_EPOCH for subsequent steps
-          echo "RUN_UNTIL_TARGET_EPOCH=$RUN_UNTIL_TARGET_EPOCH" >> "$GITHUB_ENV"
+          epoch: ${{ matrix.network.start_epoch }}
 
       - name: Run until target epoch
-        if: matrix.network.name != 'preprod' || github.ref != 'refs/heads/main'
-        timeout-minutes: ${{ fromJSON(env.TIMEOUT_RUN || inputs.default_timeout) }}
+        if: ${{ (inputs.network == '' || inputs.network == matrix.network.name) && (matrix.network.name != 'preprod' || github.ref != 'refs/heads/main') }}
+        timeout-minutes: ${{ env.DEFAULT_TIMEOUT }}
```

actionlint (1.7.12)

[error] type of expression at "float number value" must be number but found type string

(expression)